app/(main)/leaderboard/agents/AgentsLeaderboardTable.tsx

```diff
@@ -114,115 +140,133 @@ export default function AgentsLeaderboardTable() {
       order: order,
       page: 1, // Reset to first page when sorting changes
+      filter,
     });
   };
 
-  if (isLoading) return <LoadingLeaderBoardTable />;
-
   return (
     <div className="w-full">
-      <Table className="w-full">
-        <TableCaption className="text-text-low-em">
-          This Leaderboard is composed by AI trading agents betting on Omen Prediction
-          Markets contracts in gnosis chain.
-        </TableCaption>
-        <TableHeader>
-          <TableRow>
-            <TableHead>Agent</TableHead>
-            <SortableHeader
-              handleSort={handleSort}
-              sortOrder={sortOrder}
-              sortKey={sortKey}
-              headerKey={SortKey.PROFIT_LOSS}
-            >
-              Profit/Loss
-            </SortableHeader>
-            <SortableHeader
-              handleSort={handleSort}
-              sortOrder={sortOrder}
-              sortKey={sortKey}
-              headerKey={SortKey.TOTAL_VOLUME}
-            >
-              Volume Traded
-            </SortableHeader>
-            <SortableHeader
-              handleSort={handleSort}
-              sortOrder={sortOrder}
-              sortKey={sortKey}
-              headerKey={SortKey.SUCCESS_RATE}
-            >
-              Success Rate
-            </SortableHeader>
-            <SortableHeader
-              handleSort={handleSort}
-              sortOrder={sortOrder}
-              sortKey={sortKey}
-              headerKey={SortKey.TOTAL_WINS}
-            >
-              Won bets
-            </SortableHeader>
-            <SortableHeader
-              handleSort={handleSort}
-              sortOrder={sortOrder}
-              headerKey={SortKey.TOTAL_POSITIONS}
-              sortKey={sortKey}
-            >
-              Total bets
-            </SortableHeader>
-          </TableRow>
-        </TableHeader>
-        <TableBody>
-          {agentsLeaderboardData.map(agent => (
-            <TableRow key={agent.address}>
-              <TableCell>
-                <div className="flex w-64 items-center space-x-2 text-sm md:text-base">
-                  <Avatar address={agent.address} />
-                  <Link
-                    href={`/profile?address=${agent.address}`}
-                    className="truncate hover:underline"
-                  >
-                    {agent.label}
-                  </Link>
-                </div>
-              </TableCell>
-              <TableCell
-                className={twMerge(
-                  'text-right',
-                  agent.profit_loss >= 0 && 'text-text-success-main',
-                  agent.profit_loss <= -0.01 && 'text-text-danger-main'
-                )}
+      <Input
+        leftIcon="search"
+        className="mb-4 w-full"
+        placeholder="Search agent by address or label"
+        value={inputValue}
+        onChange={e => setInputValue(e.target.value)}
+      />
+      {isLoading ? (
+        <LoadingLeaderBoardTable />
```

Comment:
I think this loading is too big and the page jumps a lot when you do a search, maybe we can have a smaller loading?

Reply:
I reduced the number of rows on the LoadingTable, it's gets a bit better, but it doesn't solve the problem.